llvm/lib/ProfileData/MemProf.cpp

```diff
@@ -410,6 +410,161 @@ CallStackId hashCallStack(ArrayRef<FrameId> CS) {
   return CSId;
 }
 
+// Returns the sorted list of call stacks.
+std::vector<CallStackRadixTreeBuilder::CSIdPair>
+CallStackRadixTreeBuilder::sortCallStacks(
+    llvm::MapVector<CallStackId, llvm::SmallVector<FrameId>>
+        &MemProfCallStackData) {
+  // Create a list of call stacks to be sorted.
+  std::vector<CSIdPair> CallStacks;
+  CallStacks.reserve(MemProfCallStackData.size());
+  for (auto &[CSId, CallStack] : MemProfCallStackData) {
```

Comment:
If we expect MemProfCallStackData to be unused by the caller of build(), could it be std::move'd and then use takeVector here to avoid the copying?

Reply:
This is really a good idea!  After I switched to `takeVector()`, `sortCallStacks` doesn't have much going on, so I've folded it into its caller -- `build()`.